inference-engine/src/inference_engine/include/openvino/runtime/executable_network.hpp

```diff
@@ -0,0 +1,155 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @brief A header file that provides ExecutableNetwork class
+ *
+ * @file ie_executable_network.hpp
+ */
+
+#pragma once
+
+#include <map>
+#include <memory>
+#include <ostream>
+#include <string>
+#include <vector>
+
+#include "ie_parameter.hpp"
+#include "infer_request.hpp"
+#include "remote_context.hpp"
+
+namespace InferenceEngine {
+class IExecutableNetworkInternal;
+class RemoteContext;
+}  // namespace InferenceEngine
+namespace ngraph {
+namespace op {
+namespace v0 {
+class Parameter;
+class Result;
+}  // namespace v0
+}  // namespace op
+}  // namespace ngraph
+namespace ov {
+class Function;
+namespace runtime {
+
+class Core;
+
+/**
+ * @brief This is an interface of an executable network
+ */
+class INFERENCE_ENGINE_API_CLASS(ExecutableNetwork) {
+    std::shared_ptr<SharedObject> _so;
+    std::shared_ptr<InferenceEngine::IExecutableNetworkInternal> _impl;
+
+    /**
+     * @brief Constructs ExecutableNetwork from the initialized std::shared_ptr
+     * @param so Plugin to use. This is required to ensure that ExecutableNetwork can work properly even if plugin
+     * object is destroyed.
+     * @param impl Initialized shared pointer
+     */
+    ExecutableNetwork(const std::shared_ptr<SharedObject>& so,
+                      const std::shared_ptr<ie::IExecutableNetworkInternal>& impl);
+    friend class ov::runtime::Core;
+
+public:
+    /**
+     * @brief A default constructor.
+     */
+    ExecutableNetwork() = default;
+
+    /**
+     * @brief Get executable graph information from a device
+     *
+     * @return Function containing Executable Graph Info
+     */
+    std::shared_ptr<const Function> get_function() const;
```

Comment:
@vladimir-paramuzov @nkogteva are you OK with such new name for `GetExecGraphInfo()`? Maybe you have some alternatives?

The original idea - since ExecutableNetwork is already exec network, we don't need explicit `Exec` in `GetExecGraphInfo`. So, we want to give a more simple name

Reply:
@ilya-lavrenov I agree that  `ExecutableNetwork ` is executable itself. But I think we can confuse the user which exactly function he will get: original one or not. Maybe there is a sense to add some clarification: `get_runtime_graph` or `get_runtime_function`. Anyway, I have no serious objections.

Reply:
I like `get_runtime_function`, thanks!

Reply:
replaced with `get_runtime_function`